packages/gpac/build.sh

```diff
@@ -5,29 +5,20 @@ TERMUX_PKG_LICENSE="LGPL-2.1"
 TERMUX_PKG_MAINTAINER="@termux"
 TERMUX_PKG_VERSION="2.4.0"
 TERMUX_PKG_REVISION=1
-TERMUX_PKG_SRCURL=https://github.com/gpac/gpac/archive/refs/tags/v${TERMUX_PKG_VERSION}.tar.gz
-TERMUX_PKG_SHA256=99c8c994d5364b963d18eff24af2576b38d38b3460df27d451248982ea16157a
+TERMUX_PKG_SRCURL=(
+	https://github.com/gpac/gpac/archive/refs/tags/v"${TERMUX_PKG_VERSION}".tar.gz
+	https://github.com/gpac/gpac/commit/18863aa2176e423dae2a6d7e39ff6ed6a37b2b78.patch
```

Comment:
I think default should be to keep the patches in termux-packages, since it is then easier to inspect the patches, and they will not disappear due to changes at source site. For cases like in this example (patch from same repo as source) it seems fine though

Reply:
I think one of the primary motivation examples for this change is most likely `aubio` (5 upstream commits to implement FFmpeg 7 support - last upstream release in February 2019)

https://github.com/termux/termux-packages/blob/0cde969031671d67f28a4e23150eb10c5903b582/packages/aubio/build.sh#L13-L36

In my original implementation example of FFmpeg 7 support for `aubio` in termux-packages, I put all the patches directly copied and pasted into the `TERMUX_PKG_BUILDER_DIR`, but then when biswa96 copied and pasted my implementation into their PR, they rewrote it to be like you see in the repo now, maybe because they like how it looks this way better, but I'm not sure.

I don't know which way is objectively the better way.

Reply:
My use of these commits was originally based on their use in [Arch Linux's `aubio` package](https://gitlab.archlinux.org/archlinux/packaging/packages/aubio/-/blob/56986603eec1be74cf205b8dded14d4972cf06b7/PKGBUILD#L31-L64), 
so biswa96 appears to be attempting to rewrite/add features to termux-packages in such a way that it allows `build.sh` files to roughly mimic the layout and behavior of the Arch Linux `aubio` `PKGBUILD`. 

It seems like a good idea on the surface, but I agree that changes should be adapted to best fit the needs of  the termux-packages repository as appropriate, without necessarily trying to copy the exact same source package structure from other distros, so whichever way **you** think is best is probably the best way for termux-packages.

Reply:
Thank you for your thoughts! While I may have a slightly different perspective, I appreciate your reasoning about keeping the patches in termux-packages for easier inspection and stability. It’s definitely an important consideration.